File: lesson10/app/src/main/java/br/odb/nehe/lesson10/GL2JNILib.java
package br.odb.nehe.lesson10;

import android.content.res.AssetManager;
import android.graphics.Bitmap;

class GL2JNILib {

    public static AssetManager assetManager;

    static {
        System.loadLibrary("NdkGlue");
    }

    /**
     * @param width  the current view width
     * @param height the current view height
     */
    public static native void init(int width, int height);

    public static native void step();

    public static native void tick();

    public static native void onDestroy();

    public static native void onCreate(AssetManager assetManager);

    public static native void setTexture(Bitmap[] bitmap);

    public static native void reset();

    public static native void moveForward();

    public static native void turnRight();

    public static native void moveBackward();

    public static native void turnLeft();

    public static native void onTouchNormalized(float x, float y);
}

File: lesson10/app/src/main/java/br/odb/nehe/lesson10/GL2JNILib.kt
package br.odb.nehe.lesson10

import android.content.res.AssetManager
import android.graphics.Bitmap

internal object GL2JNILib {

	/**
	 * @param width  the current view width
	 * @param height the current view height
	 */
	external fun init(width: Int, height: Int)
	external fun step()
	external fun tick()
	external fun onDestroy()
	external fun onCreate(assetManager: AssetManager?)
	external fun setTexture(bitmap: Array<Bitmap?>?)
	external fun reset()
	external fun moveForward()
	external fun turnRight()
	external fun moveBackward()
	external fun turnLeft()
	external fun onTouchNormalized(x: Float, y: Float)

	init {
		System.loadLibrary("NdkGlue")
	}
}